From 76e2dca445e07c58845fb9695660359f0a8a51bb Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:12 -0400
Subject: [tools] perf/session: Add missing members to perf_event__attr_swap()

Message-id: <1408462094-14194-204-git-send-email-jolsa@redhat.com>
Patchwork-id: 88173
O-Subject: [PATCH RHEL7.1 BZ1131394 203/325] perf session: Add missing members to perf_event__attr_swap()
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 7db5952846dfa015d74e64b5e8656acac979490b
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Oct 18 15:29:03 2013 +0300

description
===========
The perf_event__attr_swap() method needs to swap all members of struct
perf_event_attr.  Add missing ones.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/session.c b/tools/perf/util/session.c
index 429b1f6..a9a8b34 100644
--- a/tools/perf/util/session.c
+++ b/tools/perf/util/session.c
@@ -453,6 +453,9 @@ void perf_event__attr_swap(struct perf_event_attr *attr)
  attr->bp_type  = bswap_32(attr->bp_type);
  attr->bp_addr  = bswap_64(attr->bp_addr);
  attr->bp_len  = bswap_64(attr->bp_len);
+ attr->branch_sample_type = bswap_64(attr->branch_sample_type);
+ attr->sample_regs_user  = bswap_64(attr->sample_regs_user);
+ attr->sample_stack_user  = bswap_32(attr->sample_stack_user);
 
  swap_bitfield((u8 *) (&attr->read_format + 1), sizeof(u64));
 }
-- 
1.7.1